static/app/components/feedback/feedbackItem/useFeedbackActions.ts

```diff
@@ -0,0 +1,69 @@
+import {useCallback} from 'react';
+
+import {
+  addErrorMessage,
+  addLoadingMessage,
+  addSuccessMessage,
+} from 'sentry/actionCreators/indicator';
+import useMutateFeedback from 'sentry/components/feedback/useMutateFeedback';
+import {t} from 'sentry/locale';
+import {GroupStatus} from 'sentry/types';
+import {trackAnalytics} from 'sentry/utils/analytics';
+import type {FeedbackIssue} from 'sentry/utils/feedback/types';
+import useOrganization from 'sentry/utils/useOrganization';
+
+interface Props {
+  feedbackItem: FeedbackIssue
+}
+
+const mutationOptions = {
+  onError: () => {
+    addErrorMessage(t('An error occurred while updating the feedback.'));
+  },
+  onSuccess: () => {
+    addSuccessMessage(t('Updated feedback'));
+  },
+};
+
+export default function useActions({feedbackItem}:Props) {
+  const organization = useOrganization();
+  const hasSpamFeature = organization.features.includes('user-feedback-spam-filter-ui');
+
+  const {markAsRead, resolve} = useMutateFeedback({
+    feedbackIds: [feedbackItem.id],
+    organization,
+    projectIds: [feedbackItem.project.id],
+  });
+
+  // reuse the issues ignored category for spam feedbacks
+  const isResolved = feedbackItem.status === GroupStatus.RESOLVED;
+  const isSpam = feedbackItem.status === GroupStatus.IGNORED;
+
+  return {
+    isResolved,
+    onResolveClick: useCallback(() => {
```

Comment:
nit: could make the return values a bit more readable by moving their values into `const`s up above, instead of doing all the logic in the `return` statement